### DISCUSSION TITLE
url parameter can be too short for whole-show stats

### DISCUSSION THREAD

**Discussion by jamescridland** [skymethod/op3#13] [General]
For [this example](https://op3.dev/api/1/redirect-logs?startAfter=2022-11-25T00:00:00.000Z&end=2022-11-26T00:00:00.000Z&format=json&url=https://op3.dev/e/chrt.fm/track/4EB79A/https://verifi.podscribe.com/rss/p/traffic.megaphone.fm/GLSS*&limit=1000&token=preview07ce)...

```
https://op3.dev/api/1/redirect-logs?
startAfter=2022-11-25T00:00:00.000Z&
end=2022-11-26T00:00:00.000Z
&format=json
&url=https://op3.dev/e/chrt.fm/track/4EB79A/https://verifi.podscribe.com/rss/p/traffic.megaphone.fm/GLSS*
&limit=1000
&token=preview07ce
```

... it returns an error that "destination url pathname must be at least 8 characters long, found /GLSS". As it stands, this won't be able to return "whole show" details for any podcast on Megaphone: only episode-level data. (Megaphone adds a `?updated=:unixtime` parameter on the back, incidentally).

`https://op3.dev/e/mp3s.nashownotes.com/PC20-*` is the URL for Podcasting 2.0; which again, is too short.

I did think that one way of achieving this would be to add the podcast GUID as a foolproof way to request shows; but that's not indexed, so `https://op3.dev/e,pg=9b024349-ccf0-5f69-a609-6b82873eab3c/podnews.net/audio*` also returns the same error.

Do you think there's a way of grabbing total show data in this way? Or is there a better way?

**Comment by johnspurlock-skymethod** [comment]
That pathname limitation is arbitrary, the idea was to make it somewhat selective, otherwise every query will read a ton of records for each day in the query time range (for popular hosting domains).

However, it's the holiday season, the season of giving - so I just pushed a change to production that includes effectively removing that pathname limitation altogether!

Try to limit any unselective queries like this to a single day if you can (as you've done in your sample query above)
